include/envoy/matcher/matcher.h

```diff
@@ -236,5 +237,24 @@ template <class DataType> class DataInputFactory : public Config::TypedFactory {
   }
 };
 
+class GenericDataInput {
```

Comment:
I just realised that this name could be more specific, or at least one that makes it obvious that it's not using protocol data. Is `EnvironmentalDataInput` too specific?

Reply:
Yea it's bit too... generic. :) 

I think I can imagine use cases where calling it Environmental might be a stretch (e.g. a random input string), how about `ProtocolAgnosticInput`? 

@htuch @mattklein123 for thoughts. Context here if you don't want to read the whole PR is wanting to have an extension point for data inputs that aren't templated with the protocol-specific data type (e.g. HttpMatchingData), so that certain data inputs can be shared between all different protocols (HTTP, Thrift, etc.) without duplicate factory registration per protocol.

Reply:
To help with the bike shed, maybe `CommonProtocolInputs`?

Reply:
I like the `CommonProtocolnputs` name, pushing that as a change